pkg/api/v1/router_server_secrets.go

```diff
@@ -0,0 +1,149 @@
+package serverservice
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"go.hollow.sh/serverservice/internal/dbtools"
+	"go.hollow.sh/serverservice/internal/models"
+)
+
+func (r *Router) serverSecretGet(c *gin.Context) {
+	mods := []qm.QueryMod{
+		models.ServerSecretWhere.ServerID.EQ(c.Param("uuid")),
+		qm.InnerJoin(fmt.Sprintf("%s as t on t.%s = %s.%s",
+			models.TableNames.ServerSecretTypes,
+			models.ServerSecretTypeColumns.ID,
+			models.TableNames.ServerSecrets,
+			models.ServerSecretColumns.ServerSecretTypeID,
+		)),
+		qm.Where(fmt.Sprintf("t.%s=?", models.ServerSecretTypeColumns.Slug), c.Param("slug")),
+		qm.Load(models.ServerSecretRels.ServerSecretType),
+	}
+
+	dbS, err := models.ServerSecrets(mods...).One(c.Request.Context(), r.DB)
+	if err != nil {
+		dbErrorResponse(c, err)
+		return
+	}
+
+	decryptedValue, err := dbtools.Decrypt(c.Request.Context(), r.SecretsKeeper, dbS.Value)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, &ServerResponse{Message: "error decrypting secret value", Error: err.Error()})
```

Comment:
I'd suggest not returning the decryption error and just logging it. While this may become a usability issue, this would protect from accidentally leaking sensitive information on the secret that failed to be decrypted.

Reply:
I'm open to this idea, I've checked what the errors are when I was doing some testing and the didn't include any sensitive data. Right now we have taken the approach of over returning internals on errors with this service. I'm not against no longer doing that but it is the pattern established so far.